File: java/lib/SnobotSim/MotorSim/BaseDcMotorSimulator.cpp
/*
 * BaseDcMotorSimulator.cpp
 *
 *  Created on: May 9, 2017
 *      Author: preiniger
 */


#include "SnobotSim/MotorSim/BaseDcMotorSimulator.h"


BaseDcMotorSimulator::BaseDcMotorSimulator(const DcMotorModel& aMotorModel, double aConversionFactor) :
        mMotorModel(aMotorModel), mConversionFactor(aConversionFactor), mVoltagePercentage(0)

{

}

BaseDcMotorSimulator::~BaseDcMotorSimulator()
{

}


void BaseDcMotorSimulator::SetVoltagePercentage(double aSpeed)
{
    mVoltagePercentage = aSpeed;
}

double BaseDcMotorSimulator::GetVoltagePercentage()
{
    return mVoltagePercentage;
}

double BaseDcMotorSimulator::GetVelocity()
{
    return mMotorModel.GetVelocity() * mConversionFactor;
}

double BaseDcMotorSimulator::GetPosition()
{
    return mMotorModel.GetPosition() * mConversionFactor;
}

double BaseDcMotorSimulator::GetAcceleration()
{
    return mMotorModel.GetAcceleration() * mConversionFactor;
}

void BaseDcMotorSimulator::Reset()
{
    Reset(0, 0, 0);
}

void BaseDcMotorSimulator::Reset(double aPosition, double aVelocity, double aCurrent)
{
    mMotorModel.Reset(0, 0, 0);
}

double BaseDcMotorSimulator::GetCurrent()
{
    return mMotorModel.GetCurrent();
}

File: snobot_sim_java/src/main/java/com/snobot/simulator/motor_sim/BaseDcMotorSimulator.java
package com.snobot.simulator.motor_sim;

public abstract class BaseDcMotorSimulator implements IMotorSimulator
{
    protected final DcMotorModel mMotorModel;
    protected final double mConversionFactor;
    protected double mVoltagePercentage;

    public BaseDcMotorSimulator(DcMotorModel aMotorModel)
    {
        this(aMotorModel, 1);
    }

    public BaseDcMotorSimulator(DcMotorModel aMotorModel, double aConversionFactor)
    {
        mMotorModel = aMotorModel;
        mConversionFactor = aConversionFactor;
    }

    @Override
    public void setVoltagePercentage(double aSpeed)
    {
        mVoltagePercentage = aSpeed;
    }

    @Override
    public double getVoltagePercentage()
    {
        return mVoltagePercentage;
    }

    @Override
    public double getVelocity()
    {
        return mMotorModel.getVelocity() * mConversionFactor;
    }

    @Override
    public double getPosition()
    {
        return mMotorModel.getPosition() * mConversionFactor;
    }

    @Override
    public double getAcceleration()
    {
        return mMotorModel.getAcceleration() * mConversionFactor;
    }

    @Override
    public void reset()
    {
        reset(0, 0, 0);
    }

    @Override
    public void reset(double aPosition, double aVelocity, double aCurrent)
    {
        mMotorModel.reset(aPosition, aVelocity, aCurrent);
    }

    @Override
    public double getCurrent()
    {
        return mMotorModel.getCurrent();
    }

    public DcMotorModelConfig getMotorConfig()
    {
        return mMotorModel.getMotorConfig();
    }

}
